assets/src/vo/date-time/datetime.js

```diff
@@ -0,0 +1,1156 @@
+				key,
+				value,
+				set
+			);
+			return result;
+		}, {} );
+	}
+
+	/**
+	 * Returns the date and time unit names
+	 * @return {string[]} An array of unit names
+	 */
+	[ privateMethods.getUnitNames ]() {
+		return validDateTimeUnits;
+	}
+
+	/**
+	 * Creates the various getter and setters for the value object.
+	 */
+	[ privateMethods.createGettersAndSetters ]() {
+		this[ privateMethods.getUnitNames ]().forEach(
+			( unitName ) => {
+				// creates accessor for getting the unit value via a
+				// property (eg. instance.hour)
+				Object.defineProperty( this, unitName, {
+					get() {
+						const methodName = DateTime[ privateMethods.normalizeUnitName ]( unitName );
+						const unitValue = this[ privateProperties.datetime ]
+							[ methodName ]();
+						return DateTime[ privateMethods.normalizeUnitValue ](
+							unitName,
+							unitValue,
+							false
+						);
+					},
+				} );
+				// creates a fluent setter for the value.
+				Object.defineProperty( this, 'set' + capitalize( unitName ), {
+					get() {
+						return ( value ) => {
+							return this.set( { [ unitName ]: value } );
+						};
+					},
+				} );
+			}
+		);
+	}
+
+	/**
+	 * Used to set various parts of the datetime string and returns a NEW
+	 * instance of DateTime
+	 *
+	 * Note: this will construct a DateTime even with invalid units. Make use of
+	 * `isValid()` to determine whether the instance is a valid DateTime or not.
+	 *
+	 * @param {{}} setObject An object where keys are the units.
+	 * @return {DateTime} A new instance of DateTime.
+	 */
+	set( setObject = {} ) {
+		setObject = DateTime[ privateMethods.normalizeUnitObject ]( setObject );
+		return new DateTime(
+			this[ privateProperties.datetime ]
+				.clone()
+				.set( setObject ),
+			this.timezone,
+			this.locale
+		);
+	}
+
+	/**
+	 * Accessor for the timezone string.
+	 *
+	 * @return {string} The timezone string
+	 */
+	get timezone() {
+		return this[ privateProperties.datetime ].tz();
+	}
+
+	/**
+	 * Fluent setter for the timezone property.
+	 *
+	 * @param {string} timezone
+	 * @return {DateTime} Returns a new instance of DateTime
+	 */
+	setTimezone( timezone ) {
+		DateTime.assertTimezoneIsValid( timezone );
+		return new DateTime(
+			this[ privateProperties.datetime ]
+				.clone().tz( timezone )
+		);
+	}
+
+	/**
+	 * Returns the number of days for the month set in this instance.
+	 *
+	 * @return {number}  The number of days in the month.
+	 */
+	get daysInMonth() {
+		return this[ privateProperties.datetime ].daysInMonth();
+	}
+
+	/**
+	 * Whether the current instance in time is currently in Daylight Savings
+	 * Time.
+	 *
+	 * @return {boolean} True means it is currently in Daylight Savings Time.
+	 */
+	get isInDST() {
+		return this[ privateProperties.datetime ].isDST();
+	}
+
+	/**
+	 * Whether the current instance in time is currently in a leap year.
+	 *
+	 * @return {boolean} True means this date time is in a leap year.
+	 */
+	get isInLeapYear() {
```

Comment:
> Converting that logic to a method name makes isLeapYear() the only correct option imho

A matter of personal opinion.  I personally like `isInLeapYear` because `isLeapYear` connotes the idea of checking whether the "year" value is a leap year whereas `isInLeapYear` connotes the idea of checking whether this instance in time is **in** a leap year.  So I don't think its "wrong" or "stupid" terminology at all.

Maybe think of it this way:

The date and time right now is "July 5th, 2018 11:52 am".  Is this moment in time within a Leap Year?  No it is not in a leap year.  

Remember the `DateTime` value object does not just represent a "year" it represents an instance in time.

I'm going to leave as is because of that and because both moment and luxon use this nomenclature.